### PR TITLE
Initial commit System.Reflection.

### PR DESCRIPTION
**Pull request by Priya91**

### CODE REVIEW

File: src/System.Reflection/tests/System.Reflection.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    <AssemblyName>System.Reflection.Tests</AssemblyName>
+    <RootNamespace>System.Reflection.Tests</RootNamespace>
+    <ProjectGuid>{B027C72E-F04E-42E0-A7F7-993AEF8400D2}</ProjectGuid>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the configurations -->
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+  </PropertyGroup>
+  <!-- References are resolved from packages.config -->
```
Comment (akoeplinger) [inline src/System.Reflection/tests/System.Reflection.Tests.csproj]: This line should be removed, I just cleaned those up in another PR across the board.

Comment (Priya91) [inline src/System.Reflection/tests/System.Reflection.Tests.csproj]: Yes, i'll remove this.

**Comment by stephentoub** [comment]
@Priya91, some of the tests are failing on Linux due to what looks like our current lack of proper culture support:
```
14:56:54 Starting:    System.Reflection.Tests
14:56:55 [31m   System.Reflection.Tests.CultureNameTests.SettingNullCultureNameSucceeds [FAIL]
14:56:55 [37mFinished:    System.Reflection.Metadata.Tests
14:56:55       Assert.Equal() Failure
14:56:55                 ��� (pos 0)
14:56:55       Expected: en-US
14:56:55       Actual:   
14:56:55                 ��� (pos 0)
14:56:55 Finished:    System.Reflection.TypeExtensions.Tests
14:56:55 [30m      Stack Trace:
14:56:55 [39;49m[39;49m            at System.Reflection.Tests.CultureNameTests.SettingNullCultureNameSucceeds()
14:56:55 [31m   System.Reflection.Tests.AssemblyName_ProcessorArchitectureTests.GetFullNameAndToString_AreEquivalentAndDoNotPreserveArchitecture [FAIL]
14:56:55 [37m      Assert.Equal() Failure
14:56:55                               ��� (pos 14)
14:56:55       Expected: Test, Culture=en-US, PublicKeyToken=b77a5c561934e089
14:56:55       Actual:   Test, Culture=neutral, PublicKeyToken=b77a5c561934e089
14:56:55                               ��� (pos 14)
14:56:55 [30m      Stack Trace:
14:56:55 [37m            at System.Reflection.Tests.AssemblyName_ProcessorArchitectureTests.GetFullNameAndToString_AreEquivalentAndDoNotPreserveArchitecture()
14:56:55 [31m   System.Reflection.Tests.CultureNameTests.SettingEmptyCultureNameSucceeds [FAIL]
14:56:55 [37m      Assert.Equal() Failure
14:56:55                 ��� (pos 0)
14:56:55       Expected: en-US
14:56:55       Actual:   
14:56:55                 ��� (pos 0)
14:56:55 [30m      Stack Trace:
14:56:55 [37m            at System.Reflection.Tests.CultureNameTests.SettingEmptyCultureNameSucceeds()
14:56:55 [31m   System.Reflection.Tests.CultureNameTests.SettingValidCultureNameSucceeds [FAIL]
14:56:55 [37m      Assert.Equal() Failure
14:56:55                 ��� (pos 0)
14:56:55       Expected: en-US
14:56:55       Actual:   
14:56:55                 ��� (pos 0)
14:56:55 [30m      Stack Trace:
14:56:55 [37m            at System.Reflection.Tests.CultureNameTests.SettingValidCultureNameSucceeds()
14:56:55 [31m   System.Reflection.Tests.CultureNameTests.SettingCultureNameIsCaseInsensitive [FAIL]
14:56:55 [37m      Assert.Equal() Failure
14:56:55                 ��� (pos 0)
14:56:55       Expected: en-US
14:56:55       Actual:   
14:56:55                 ��� (pos 0)
14:56:55 [30m      Stack Trace:
14:56:55 [37m            at System.Reflection.Tests.CultureNameTests.SettingCultureNameIsCaseInsensitive()
14:56:55 [31m   System.Reflection.Tests.CultureNameTests.SettingInvalidCultureNameThrowsCultureNotFound [FAIL]
14:56:55 [37m      Assert.Throws() Failure
14:56:55       Expected: typeof(System.Globalization.CultureNotFoundException)
14:56:55       Actual:   (No exception was thrown)
14:56:55 [30m      Stack Trace:
14:56:55 [37m            at System.Reflection.Tests.CultureNameTests.SettingInvalidCultureNameThrowsCultureNotFound()
14:56:55 Finished:    System.Reflection.Tests
14:56:55 [37m
14:56:55 === TEST EXECUTION SUMMARY ===
14:56:55 [37m   System.Reflection.Tests  Total: 657, Errors: 0, Failed: 6, Skipped: 0, Time: 1.227s
```
I suggest you double check that's what's actually happening, and then suppress these tests on Unix with:
```
[ActiveIssue(846, PlatformID.AnyUnix)]
```

**Comment by Priya91** [comment]
@stephentoub : I am in the process of pushing that change. :)

**Comment by stephentoub** [comment]
Thanks.  Looks like one got missed:
```
16:36:43 [31m   System.Reflection.Tests.CultureNameTests.SettingNullCultureNameSucceeds [FAIL]
16:36:43 [37m      Assert.Equal() Failure
16:36:43                 ��� (pos 0)
16:36:43       Expected: en-US
16:36:43       Actual:   
16:36:43                 ��� (pos 0)
16:36:43 [39;49m[30m      Stack Trace:
16:36:43 [39;49m            at System.Reflection.Tests.CultureNameTests.SettingNullCultureNameSucceeds()
```

**Comment by Priya91** [comment]
retest this please

**Pull request merged**